contracts/exchange/util/BIMath.sol

```diff
@@ -98,4 +102,43 @@ library BIMath {
     }
     return SafeCast.toUint64(SafeCast.toUint256(c));
   }
+
+  function zero() internal pure returns (BI memory) {
+    return BI(0, 0);
+  }
+
+  function one() internal pure returns (BI memory) {
+    return BI(1, 0);
+  }
+
+  function half() internal pure returns (BI memory) {
+    return BI(int256(5), 1);
+  }
+
+  function floor(BI memory b) internal pure returns (BI memory) {
+    return scale(scale(b, 0), b.dec);
+  }
+
+  function roundDown(BI memory b) internal pure returns (BI memory) {
+    if (b.val > 0) {
+      return floor(b);
+    }
+    return floor(sub(b, one()));
+  }
+
+  function roundUp(BI memory b) internal pure returns (BI memory) {
```

Comment:
can you write a foundry test that replicate bi_test.go, focusing on round* methods?

Reply:
Done!